frontend/TestRun/ArtifactTab.svelte

```diff
@@ -10,6 +10,30 @@
     let logName = $state("");
     let logLink = $state("");
     const dispatch = createEventDispatcher();
+    const DOWNLOAD_DELAY_MS = 800
+
+    const getArtifactDownloadLink = (logName: string) =>
+        `/api/v1/tests/${testInfo.test.plugin_name}/${testRun.id}/log/${logName}/download`;
+
+    const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+    const downloadAllArtifacts = async () => {
+        if (!testRun?.logs?.length) return;
+        if (typeof document === "undefined") return;
+
+        for (const [logName] of testRun.logs) {
+            const downloadUrl = getArtifactDownloadLink(logName);
+            const linkElement = document.createElement("a");
+            linkElement.href = downloadUrl;
+            linkElement.setAttribute("download", "");
+            linkElement.style.display = "none";
+            document.body.appendChild(linkElement);
+            linkElement.click();
+            document.body.removeChild(linkElement);
+
+            await delay(DOWNLOAD_DELAY_MS);
```

Comment:
Could dispatch `setTimeout` with increasing timeout for each link instead:

```js
let delay = 0;
for ... {
   setTimeout(click, delay);
   delay += DOWNLOAD_DELAY_MS;
}
```
In any case it would be good to block download buttton until all delays were fired, whether with "blocking" loop like this or stored promises - just add a final callback that would unblock the button and lock it at the start of the function.